pkg/ctl/cmdutils/configfile.go

```diff
@@ -619,20 +619,10 @@ func NewGetIAMServiceAccountLoader(cmd *Cmd, sa *api.ClusterIAMServiceAccount) C
 	}
 
 	l.validateWithoutConfigFile = func() error {
-		sa.AttachPolicyARNs = []string{""} // force to pass general validation
-
 		if l.ClusterConfig.Metadata.Name == "" {
 			return ErrMustBeSet(ClusterNameFlag(cmd))
 		}
 
-		if l.NameArg != "" {
```

Comment:
Does this remove support for `--name` as positional argument?

Reply:
`name` and `namespace` can now be provided without each other being specified. The PR description has some examples

Reply:
does `./eksctl get iamserviceaccount --cluster jk foo` work? (i think it worked before)

Reply:
yeah, thats the first example output in the PR description 

```
$ ./eksctl get iamserviceaccount --cluster jk
[ℹ]  eksctl version 0.37.0-dev+a487044b.2021-01-22T14:30:12Z
[ℹ]  using region us-west-2
NAMESPACE       NAME    ROLE ARN
baz             baz     arn:aws:iam::<redcated>:role/eksctl-jk-addon-iamserviceaccount-baz-baz-Role1-A1PVCNCBI0WX
baz             foo     arn:aws:iam::<redcated>:role/eksctl-jk-addon-iamserviceaccount-baz-foo-Role1-1SDQ5H4X68B2P
default         bar     arn:aws:iam::<redcated>:role/eksctl-jk-addon-iamserviceaccount-default-ba-Role1-1CPB5MICLNMXR
default         foo     arn:aws:iam::<redcated>:role/eksctl-jk-addon-iamserviceaccount-default-fo-Role1-SUWVUTGQKJX7
```

Reply:
in general all `eksctl` commands work with a _positional_ `name` argument, so `--cluster jk foo`

Reply:
ohhh sorry I completely didn't see the `foo` after the cluster name, my bad. I didn't know that was a thing.

Trying it out with an older version of the CLI it doesn't appear to work:
```
eksctl get iamserviceaccount --cluster jk foo
NAMESPACE       NAME                    ROLE ARN
backend-apps    cache-access            arn:aws:iam::<redacted>:role/eksctl-jk-addon-iamserviceaccount-backend-ap-Role1-LQ6DV2ILV1NM
backend-apps    s3-reader               arn:aws:iam::<redacted>:role/eksctl-jk-addon-iamserviceaccount-backend-ap-Role1-ZTYY4YBJTYX6
baz             baz                     arn:aws:iam::<redacted>:role/eksctl-jk-addon-iamserviceaccount-baz-baz-Role1-A1PVCNCBI0WX
baz             foo                     arn:aws:iam::<redacted>:role/eksctl-jk-addon-iamserviceaccount-baz-foo-Role1-1SDQ5H4X68B2P
```

Reply:
is this coming back then?

Reply:
IMO we can leave positional name out if it didn't work before, although i'd prefer this to use the "same" machinery for the CLI args the rest of the code uses with a best effort (or refactor the existing where necessary), which may mean bring positional name back